gtsam/linear/NoiseModel.cpp

```diff
@@ -774,6 +799,16 @@ Cauchy::Cauchy(double k, const ReweightScheme reweight) : Base(reweight), k_(k),
   }
 }
 
+double Cauchy::weight(double error) const {
+  return ksquared_ / (ksquared_ + error*error);
+}
+
+double Cauchy::residual(double error) const {
+  const double xc2 = error / k_;
```

Comment:
minor nit: `xc2` is usually `error*error/ksquared_`, and then can do `std::log(1 + xc2)` below.

Reply:
@varunagrawal I know you can't finish this yet because of other priorities, but could you incorporate this change into the new PR?